lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -2,11 +2,30 @@ module ActiveModelSerializers
   module Adapter
     class JsonApi
       class ResourceIdentifier
+        def self.type_for(class_name, serializer_type = nil, transform_options = {})
```

Comment:
reason for making a class method: tied too much logic to serializer instances make refactoring difficult, especially when, as demonstrated here, the actual serializer instance is not necessary for the transformation

see problems in https://github.com/rails-api/active_model_serializers/pull/1857 :)